driver/src/main/scala/api/commands/AtlasSearchAggregation.scala

```diff
@@ -1079,18 +1115,23 @@ private[commands] trait AtlasSearchAggregation[P <: SerializationPack] {
        * import reactivemongo.api.bson.collection.BSONCollection
        *
        * def prepare(coll: BSONCollection): coll.AggregationFramework.AtlasSearch.Near = coll.AggregationFramework.AtlasSearch.Near(
-       *   query = "foo",
+       *   origin = Near.origin(1),
        *   path = "field",
        *   pivot = Some(0.5D)
        * )
        * }}}
        */
       def apply(
-        query: SearchString,
+        origin: Near.Origin,
         path: SearchString,
         pivot: Option[Double] = None,
         score: Option[Score] = None): Near =
-        new Near(query, path, pivot, score)
+        new Near(origin, path, pivot, score)
+
+      final class Origin private[api] (val value: pack.Value)
```

Comment:
Date (`Instant`) or Number (`T: Numeric`)

Reply:
Didn't get the point, if Writer and origin are encapsulated in `api`, then you can create Origin instance outside only with 'origin' method but you can use only writers that exist in Writer object only. But there we have only numeric and instant writers

Reply:
`Value` offers no such type safety and should only be used for serialised representation, not in API when possible.